src/Raven.Client/Documents/Operations/ETL/Transformation.cs

```diff
@@ -31,12 +28,164 @@ public class Transformation
+
+                if (IsAddingCounters && type == EtlType.Sql)
+                    errors.Add("Adding counters isn't supported by SQL ETL");
+
+                FillCollectionToLoadCounterBehaviorFunction(errors, type);
+            }
+
+            private void FillCollectionToLoadCounterBehaviorFunction(List<string> errors, EtlType type)
+            {
+                var counterBehaviors = LoadBehaviorMethodRegex.Matches(_parent.Script);
+                if (counterBehaviors.Count == 0) 
+                    return;
+            
+                if (type == EtlType.Sql)
+                {
+                    errors.Add("Load counter behavior functions aren't supported by SQL ETL");
+                    return;
+                }
+
+                CollectionToLoadCounterBehaviorFunction = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                foreach (Match counterBehaviorFunction in counterBehaviors)
+                {
+                    if (counterBehaviorFunction.Groups.Count != 2)
+                    {
+                        errors.Add(
+                            "Invalid load counters behavior function. It is expected to have the following signature: " +
+                            "loadCountersOf<CollectionName>Behavior(docId, counterName) and return 'true' if counter should be loaded to a destination");
+                    }
+
+                    var functionSignature = counterBehaviorFunction.Groups[0].Value;
+                    var collection = counterBehaviorFunction.Groups[1].Value;
+
+                    var functionName = LoadBehaviorMethodNameRegex.Match(functionSignature);
+
+                    if (_parent.Collections.Contains(collection) == false)
+                    {
+                        var scriptCollections = string.Join(", ", _parent.Collections.Select(x => ($"'{x}'")));
+
+                        errors.Add(
+                            $"There is '{functionName}' function defined in '{_parent.Name}' script while the processed collections " +
+                            $"({scriptCollections}) doesn't include '{collection}'. " +
+                            "loadCountersOf<CollectionName>Behavior() function is meant to be defined only for counters of docs from collections that " +
+                            "are loaded to the same collection on a destination side");
+                    }
+
+                    CollectionToLoadCounterBehaviorFunction[collection] = functionName.Value;
+                }
+            }
+        }
+
+        internal readonly TimeSeriesTransformation TimeSeries;
+        internal class TimeSeriesTransformation
+        {
+            internal const string Marker = "timeSeries/";
+            
+            internal static class AddTimeSeries
+            {
+                internal const string Name = "addTimeSeries";
+                public const string Signature  = "addTimeSeries(timeSeriesName, from, to)";
```

Comment:
Need to review this against the other JS API we have on timeseries, I'm pretty sure we aren't using `TimeSeries`

Reply:
In script runner it is `timeseries(doc, name).get(from, to)`
So I will modify it as follow:
```
var person = loadToPeople({ Name: this.Name + ' ' + this.LastName });
var ts = timeseries(name).get(from, to);
person.timeseries(name).append(ts);
```
Does that make sense?

Reply:
How does it relate to counters?

Reply:
In the ETL?
We do it that way:
```
var person = loadToPeople({ Name: this.Name + ' ' + this.LastName });

person.addCounter(loadCounter('down'));
```

Reply:
So what should the API be in order to have a consistent expereince?

Reply:
Consistent with other time-series JS API and ETL counters?
Maybe as follow:
```
var person = loadToPeople({ Name: this.Name + ' ' + this.LastName });
var ts = timeseries(name).load(from, to);
person.addTimeseries(ts);
```

Reply:
I don't really like it, but not sure what I would rather have here.

@karmeli87 @aviv86

Reply:
I think we should have consistent API across ETL scripts - it uses 'load' naming convention which we obviously don't use in JS patching. I prefer:

```
var ts = loadTimeSeries(timeSeriesName, from, to)
person.addTimeseries(ts);
```

instead of 

```
var ts = timeseries(name).load(from, to);
person.addTimeseries(ts);
```

because:
- we won't have so many methods that we need to introduce `timeseries(name).` wrapper
- it's consistent with handling counters in ETL scripts (`person.addCounter(loadCounter('down'));`

Reply:
I like this